core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -228,11 +232,17 @@ class AddPartitionsToTxnManager(
                 val tp = new TopicPartition(topicResult.name, partitionResult.partitionIndex)
                 if (partitionResult.partitionErrorCode != Errors.NONE.code) {
                   // Producers expect to handle INVALID_PRODUCER_EPOCH in this scenario.
-                  val code =
+                  var code =
                     if (partitionResult.partitionErrorCode == Errors.PRODUCER_FENCED.code)
                       Errors.INVALID_PRODUCER_EPOCH.code
                     else
                       partitionResult.partitionErrorCode
+                  // For backward compatibility with clients.
```

Comment:
nice 👍

Reply:
agreed. thanks for the suggestion
I will update the code here